rust/ffi/node/src/lib.rs

```diff
@@ -156,6 +159,19 @@ fn get_aws_creds<T>(
     }
 }
 
+/// Get AWS region arguments from the context
+fn get_aws_region<T>(cx: &mut FunctionContext, arg_location: i32) -> Result<Option<String>, NeonResult<T>> {
+    let region = cx
+        .argument_opt(arg_location)
+        .map(|arg| arg.downcast_or_throw::<JsString, FunctionContext>(cx));
+
+    match region {
+        Some(Ok(region)) => Ok(Some(region.value(cx))),
+        None => Ok(None),
+        Some(Err(e)) => Err(Err(e))
```

Comment:
You're returning `Err(Err(e))`.  What is this?  I would have thought the return would have been `Err(e)`.

Reply:
Yeah, I think there might be ways to simplify this. Let me try a few things.